sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -119,4 +121,202 @@ object IntervalUtils {
       case _: IllegalArgumentException => null
     }
   }
+
+  private val yearMonthPattern = Pattern.compile("^([+|-])?(\\d+)-(\\d+)$")
```

Comment:
This can just use Scala regex primitives instead.

Reply:
This will require rewriting of dependent code but I wanted to just move methods and keep them closer to original methods as much as possible. Maybe we will refactor the code in a follow up PR?

Reply:
You're already moving the code and converting to scala in other ways, so I think it's not a big deal to do it here rather than make another one for pretty minor adjustments. Or is there an argument that git won't really track the history of this code after move if it's changed a lot?

Reply:
To use Scala regexp, I will have to restructure the code significantly, so, it will be impossible to track anything.

Reply:
Why does it need any restructuring? `"...".r` gives you an object with pretty much the same API. You don't have to use regex pattern matching in Scala if it's disruptive. Well, I don't feel strongly about it, just doesn't seem any more of a change than you've already made in adapting this code to Scala.